pallets/parachain-staking/src/lib.rs

```diff
@@ -724,19 +847,52 @@ pub mod pallet {
 			Perbill,
 			Perbill,
 		),
+		/// Migrated NominatorState -> NominatorState2, ExitQueue -> ExitQueue2
+		DelayNominationExitsMigrationExecuted,
+	}
+
+	/// Storage migration for delaying nomination exits and revocations
+	fn delay_nomination_exits_migration_execution<T: Config>() {
+		if !<DelayNominationExitsMigration<T>>::get() {
+			// migrate from Nominator -> Nominator2
+			for (acc, nominator_state) in NominatorState::<T>::drain() {
+				let state: Nominator2<T::AccountId, BalanceOf<T>> = nominator_state.into();
```

Comment:
@notlesh This `.into()` uses the `From` impl, which contains exactly how we convert each `Nominator` -> `Nominator2`

Reply:
While this is an elegant use of the From, IF this is only used here in the migration, I would (for future times) argue that keeping the mutation logic directly here would benefit the reader and prevent bug where someone would use the wrong structure because it gets converted.

Reply:
That would imply we keep the old struct in the code as well. I'd prefer to move the old struct and the `From` impl to docs, something that isn't code. We can link the code at each migration commit so someone could write `From<Nominator1> for Nominator3` if they want to, but we don't have to write that ourselves.

Reply:
Sounds good